doc/user/content/transform-data/temporal-filters.md

```diff
@@ -0,0 +1,368 @@
+
+Materialize leans hard into the ideal that SQL is what you know best, and what you want to use to look at streaming data. At the same time, there are several tantalizing concepts that native stream processors provide that aren't obviously possible with standard SQL.
+
+In this article, we'll look at how to perform time-windowed computation over temporal data.
+
+## Temporal Data
+
+Temporal databases are a pretty neat thing, and we aren't going to go deep on that today. Instead, let's just start with a relation that has a pretty simple schema, and which includes some times.
+
+```sql
+-- Making a new data source.
+CREATE TABLE events (
+   content text,
+   insert_ts numeric,
+   delete_ts numeric
+);
+```
+
+We have some content, clearly you could fluff that up to something more interesting, and two additional columns. The `insert_ts` and `delete_ts` columns will play the role of timestamps, indicating when an event should first appear and when it should be removed.
+
+Of course, this is all just data right now. These are fields, and you could put whatever you want in them. They don't have any specific meaning yet. You could make `delete_ts` be smaller than `insert_ts`, if you are that sort of person.
+
+## Time-Windowed Queries
+
+The question is now what do you do with these data?
+
+In Materialize, you can ask questions that grab the "logical time" for your query (roughly: the wall clock when you run the query) and then use that in your query's logic. For example, this query counts up `content` based on events that start by and end after the current logical timestamp.
+
+```sql
+-- Reduce down to counts within a time interval.
+SELECT content, count(*)
+FROM events
+WHERE mz_now() >= insert_ts
+  AND mz_now() < delete_ts
+GROUP BY content;
+```
+
+This query will change its results over time. Not just because you might add to and remove from `events`, but because `mz_now()` advances as you stare at your screen. Each time you issue the query you may get a different result.
+
+This looks like a great query! What's not to like?
+
+The main issue really is that it is ***just a query***. You can ask this question over and over, but you can also ask the same thing with an arbitrary timestamp in place of `mz_now()`. To support that, Materialize has to keep the entire collection of data around. Your `events` table will grow and grow and grow, and the time to answer the query will grow as well.
+
+## Time-Windowed Computation
+
+Materialize specializes at ***maintaining*** computations like the above, both because that can make it faster to get your answers out, but also because by specifying what you actually need Materialize can run much more lean. We'll see that now with the query above! Moreover, comparing parts of your data to `mz_now()` in views (rather than just in queries) introduces powerful new idioms, ones that we'll explore in this article.
```

Comment:
Drive-by, but potentially "this article" may want to evolve to something else if this is landing in docs. @kay-kim to tell us how best to reference it!